### DISCUSSION TITLE
New local onedrive - Ensuring consistent local states

### DISCUSSION THREAD

**Discussion by AlbertRapp** [abraunegg/onedrive#1958] [Q&A] [answered]
I have just set up my OneDrive on Ubuntu for the first time. To save time, I have copied my files to an external hard drive and copied the same data to my Ubuntu partition then. So, the data should be available locally.
Now, I am having severe issues with the time it takes to upload a single small file. When I add a small text file (couple of KB) to my local directory, then sometimes it is uploaded within seconds but sometimes it literally takes 10 to 20 minutes for the sync to finish.
When it takes so long, then manual sync shows that it seems like the client is rechecking many, many files like so

```
Processing 249 OneDrive items to ensure consistent local state
Processing 223 OneDrive items to ensure consistent local state
Processing 263 OneDrive items to ensure consistent local state
Processing 232 OneDrive items to ensure consistent local state
....many more rows like this
```

Is this related to the fact that I just recently set up my OneDrive on Ubuntu? I would guess that by now, the client knows my data as it has been sitting in the directory for some time now. It feels weird that the client sometimes checks TONS of files when I am waiting for my one file to be uploaded. Is this something that can be changed by a setting?

**Comment by abraunegg** [comment, marked as answer]
@AlbertRapp
> I have just set up my OneDrive on Ubuntu for the first time

Hopefully you have installed the client as per https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md and then read https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages

Just in case - what application version are you using - `onedrive --version` ? The *correct* version of the client for Ubuntu should be [![Version](https://img.shields.io/github/v/release/abraunegg/onedrive)](https://github.com/abraunegg/onedrive/releases). Anything lower is obsolete and unsupported.

Additionally, what is your application configuration - `onedrive --display-config` ?

> Now, I am having severe issues with the time it takes to upload a single small file. When I add a small text file (couple of KB) to my local directory, then sometimes it is uploaded within seconds but sometimes it literally takes 10 to 20 minutes for the sync to finish.

This is *normal* application behaviour. Depending on your application version however, this behaviour can be different. From your description it sounds like you are running an older version of the client (thus being performed more often), as the latest version includes a 'major tweak' on when this integrity check is performed. Please read: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#monitor_fullscan_frequency

Additionally, do not add or use `--resync` unless the application advises you to add it due to your configuration changing ... if your application is constantly fetching all items from onedrive, this is an indicator that the known sync state has been removed or a full sync has been requested.
